client/src/redux/reducers/onboarding.reducers.js

```diff
@@ -0,0 +1,132 @@
+export const initialState = {
+  admin: {
+    users: [],
+    loading: false,
+    error: null
+  },
+  user: {
+    username: null,
+    firstName: null,
+    lastName: null,
+    setupComplete: false,
+    steps: [],
+    loading: false,
+    error: null
+  },
+  action: {
+    step: null,
+    action: null,
+    loading: false,
+    username: null,
+    error: null
+  }
+};
+
+export function updateUserFromEvent(user, event) {
+  const result = { ...user };
+  if (result.username === event.username) {
+    if (event.step === 'portal.apps.onboarding.execute.execute_setup_steps') {
```

Comment:
It sends the SetupStep class name up in the general case of step events. I don't really have a great reason for the setupComplete event full package name since it's artificial other than consistency